tests/test_create.py

```diff
@@ -1063,6 +1063,42 @@ def side_effect(self, url, **kwargs):
                 # The mock should have been called with our local channel URL though.
                 assert result_dict.get('local_channel_seen')
 
+    def test_create_from_extracted(self):
+        # Test that we can link a package from its extracted directory
+        # even if the tarball is no longer available.
+        pkgs_dir = PackageCache.first_writable().pkgs_dir
+        mkdir_p(pkgs_dir)
+        pkgs_dir_hold = pkgs_dir + '_hold'
```

Comment:
Someday we'll make all of these integration tests be able to run concurrently.  Renaming this directory here could be a problem for that.  Instead of renaming, how about overriding the pkgs_dirs config and pointing to a temporary directory just for this test?

Start the test with

    with make_temp_env() as prefix:
        pkgs_dir = join(prefix, 'pkgs')
        mkdir_p(pkgs_dir)
        touch(join(pkgs_dir, PACKAGE_CACHE_MAGIC_FILE))

        def pkgs_dir_has_tarball(tarball_startswith):
            return any(basename(f).startswith(tarball_startswith)
                       for f in glob(join(pkgs_dir, CONDA_TARBALL_EXTENSION)))

        with env_var('CONDA_PKGS_DIR', pkgs_dir, reset_context):
            # First, make sure the openssl package is present in the cache, downloading if necessary
            assert not pkgs_dir_has_tarball("openssl-")
            run_command(Commands.INSTALL, prefix, "openssl")
            run_command(Commands.REMOVE, prefix, "openssl")
            assert pkgs_dir_has_tarball("openssl-")

            # Then, remove the tarball but keep the extracted directory around
            run_command(Commands.CLEAN, prefix, "--tarballs --yes")
            assert not pkgs_dir_has_tarball("openssl-")
            assert any(d.startswith("openssl-") for d in listdir(pkgs_dir))

            # Finally, install openssl, enforcing the use of the extracted package
            # We expect that the tarball did not appear again because we simply linked the environment 
            # from the extracted directory. If the tarball appeared again, we decided to re-download 
            # the package for some reason.
            run_command(Commands.INSTALL, prefix, "openssl --offline")
            assert not pkgs_dir_has_tarball("openssl-")

            # double check that the only package cache we used was pkgs_dir
            assert context.pkgs_dir == pkgs_dir,

All of that completely untested :)

Using openssl here instead of python.  Trying to be more careful about packages used for these integration tests.  `openssl` is lighter than python with no dependencies.  It's still >2MB though.  Feel free to pick a different package.  Just needs to be one that's universal across the five primary subdirs right now (linux-64, linux-32, osx-64, win-64, win-32).